rfcs/2022-03-15-11851-ingest-opentelemtry-traces.md

```diff
@@ -0,0 +1,368 @@
+    All the expected behaviours regarding the kind of requests/responses code and sequence are clearly defined as well
+    as the default URL path (`/v1/traces` for traces).
+- Internal traces representation/normalization, two options are opened see [outstanding
+  questions](#outstanding-questions), but the consensus is leaning towards a new dedicated container that would:
+  - Move away from from the current implementation that relies on `LogEvent` to a dedicated container, the
+    implementation will stay  in [./lib/vector-core/src/event/trace.rs][current-trace-in-vector]
+  - Borrow most of its semantic from the Opentelemetry Traces format, the `TraceEvent` would then evolve toward a
+    concrete dedicated container and contains a slice of spans, the span being directly inspired by the Opentelemtry
+    specification.
+  - Some top-level information (Like trace ID, trace-wide tags/metrics, the original format)
+  - Trace would not get native `VrlTarget` representation anymore, there is a bigger discussion there that should
+    probably be adressed separately. As an interim measure few fields may be exposed (like trace ID & trace-wide
+    tags).
+- APM stats computation:
+  - Implement a similar logic that the one done in the Datadog OTLP exporter, this would allow user to use multiple
+    Datadog product with Opentelemetry traces and get the same consistent behaviour in all circumstances. APM stats
+    computation is hooked [there][apm-stats-computation] in the Datadog exporter. But as this is go code it relies on
+    the [Agent codebase][agent-code-for-otlp-exporter] to do the [actual computation][agent-handle-span].
+  - Where the APM stats computations is still under discusion, see [outstanding questions](#outstanding-questions)
+
+## Rationale
+
+- Opentelemetry is the de-facto standard for traces, so supporting it at some point is mandatory. Note that this
+  consideration is wider than just traces as metrics (and logs) are addressed by the Opentelemetry project.
+
+## Drawbacks
+
+N/A
+
+## Prior Art
+
+N/A
+
+## Alternatives
+
+- We could keep the Datadog trace-agent as an OTLP->Datadog traces converter and ingest datadog traces from there
+- We could keep the Datadog exporter as an OTLP->Datadog traces converter and ingest datadog traces from there
+- We could write a Vector exporter for the Opentelemetry collector, note that this would likely leverage the Vector
+  protocol and this logic could be applied to metrics as well
+
+## Outstanding Questions
+
+- Do we want to have a single `opentelemetry` source with names output or multiple sources for Opentelemetry metrics and
+  Opentelemetry logs, this is more a user experience discussion as this doesn't have big implementation impact.
+- APM stats computation:
+  - Either in all traces sources (to be done for each source, except for the `datadog_agent` sources where APM stats may
+    be decoded from received payloads) - likely to be the preferred solution
```

Comment:
I agree, those metrics can be useful outside datadog, I think this can be generic enough especially when we account for the trace normalisation effort. Also I think this stats computation should come with an opt in/out flag to enable/disable it upon user needs.